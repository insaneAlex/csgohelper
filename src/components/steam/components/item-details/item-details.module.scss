@import 'styles/variables.scss';

.details {
  max-width: 500px;
  margin: 0 auto;
  text-align: center;
  padding: 0 25px 25px 25px;

  .exterior {
    margin-top: 0;
    margin-bottom: 15px;
  }

  @include body-short;
}
