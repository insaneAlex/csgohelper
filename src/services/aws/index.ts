export {awsServices} from './aws';
export type {PricesType, InventoryCacheType, NoPriceInventory} from './types';
