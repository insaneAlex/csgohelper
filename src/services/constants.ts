export const AWS_REGION = 'eu-central-1';

export const INVENTORY_TABLE = 'inventories';
