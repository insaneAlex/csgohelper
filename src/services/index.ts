export {AWS_REGION, INVENTORY_TABLE} from './constants';
export {isClient} from './is-client';
export {storage} from './storage';
export {noop} from './noop';
export type {PriceType} from './types';
