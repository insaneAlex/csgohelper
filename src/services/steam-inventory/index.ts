export {inventoryApi} from './inventory';

export type {InventoryItemType, InventoryGlobalType, TagsTypeIndex} from './types';
