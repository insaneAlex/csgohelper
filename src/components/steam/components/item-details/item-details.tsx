import {inventoryImageBaseUrl} from '../../constants';
import {InventoryItemType} from '@/types';
import Image from 'next/image';
import {FC} from 'react';

import styles from './item-details.module.scss';

export const ItemDetails: FC<{item: InventoryItemType}> = ({item}) => {
  const {icon_url, name, exterior} = item;

  return (
    <section className={styles.details}>
      <h1>{name}</h1>
      {exterior && <h2 className={styles.exterior}>{`(${exterior})`}</h2>}
      <Image src={inventoryImageBaseUrl + icon_url} alt={name} unoptimized width={256} priority height={198} />
    </section>
  );
};
