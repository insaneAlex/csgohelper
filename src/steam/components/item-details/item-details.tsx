import {InventoryItemType} from "@/types";
import {FC} from "react";
import Image from "next/image";
import {inventoryImageBaseUrl} from "@/api/constants";
import {getTagValue} from "../../helpers/get-tag-value";

import styles from "./item-details.module.scss";

type Props = {item: InventoryItemType};

export const ItemDetails: FC<Props> = ({item}) => {
  const {icon_url, name, descriptions, tags} = item;
  const exterior = getTagValue({tag: "Exterior", tags});
  const imgSrc = `${inventoryImageBaseUrl}${icon_url}`;

  const renderDescriptions = () => (
    <div className={styles.descriptions}>
      {descriptions.map(({value, color}, i) => {
        return (
          value && (
            <div key={i} style={color ? {color: `#${color}`} : undefined}>
              {value}
            </div>
          )
        );
      })}
    </div>
  );
  return (
    <div className={styles.details}>
      <h1>{name}</h1>
      {exterior && <h2 className={styles.exterior}>{`(${exterior})`}</h2>}
      <Image
        src={imgSrc}
        alt={name}
        quality={100}
        width={256}
        priority
        height={198}
      />
      <hr className={styles.line} />

      {renderDescriptions()}
    </div>
  );
};
