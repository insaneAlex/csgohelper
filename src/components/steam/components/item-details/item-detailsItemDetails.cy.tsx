import React from 'react';
import {ItemDetails} from './item-details';
import {InventoryItemType} from '@/src/services/steam-inventory';

describe('<ItemDetails />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-react
    cy.mount(<ItemDetails item={{icon_url: 'test', name: 'M4A1', exterior: 'Factory new'} as InventoryItemType} />);
  });
});
