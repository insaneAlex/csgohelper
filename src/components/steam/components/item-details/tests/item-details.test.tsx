import {render} from '@testing-library/react';
import {ItemDetails} from '../item-details';
import {InventoryItemType} from '@/src/services/steam-inventory';

jest.mock('../../../../../services', () => ({}));

describe('ItemDetails', () => {
  const item = {icon_url: 'iconUrl', name: 'itemName', exterior: 'fine condition'} as InventoryItemType;
  it('should render correctly', () => {
    const {container} = render(<ItemDetails item={item} />);
    expect(container).toMatchSnapshot();
  });
});
