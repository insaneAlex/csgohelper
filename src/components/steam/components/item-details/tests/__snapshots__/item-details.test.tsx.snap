// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails should render correctly 1`] = `
<div>
  <section
    class="details"
  >
    <h1>
      itemName
    </h1>
    <h2
      class="exterior"
    >
      (fine condition)
    </h2>
    <img
      alt="itemName"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="198"
      src="https://steamcommunity-a.akamaihd.net/economy/image/iconUrl"
      style="color: transparent;"
      width="256"
    />
    <hr
      class="separator"
    />
  </section>
</div>
`;
